pkg/okteto/preview.go

```diff
@@ -337,13 +295,35 @@ func (c *previewClient) GetResourcesStatusFromPreview(ctx context.Context, previ
 	return status, nil
 }
 
-func translatePreviewAPIErr(err error, name string) error {
+func (*previewClient) translateErr(err error, name string) error {
 	if err.Error() == "conflict" {
-		return fmt.Errorf("preview '%s' already exists with a different scope. Please use a different name", name)
+		return previewConflictErr{name: name}
 	}
 	if strings.Contains(err.Error(), "operation-not-permitted") {
-		return oktetoErrors.UserError{E: fmt.Errorf("you are not authorized to create a global preview env"),
+		return oktetoErrors.UserError{E: ErrUnauthorizedGlobalCreation,
 			Hint: "Please log in with an administrator account or use a personal preview environment"}
 	}
 	return err
 }
+
+// validateNamespace checks if the new ns exceeds the max chars allowed by k8s and if t
+func validateNamespace(namespace, object string) error {
```

Comment:
To clean things even further we could remove the global var `MAX_ALLOWED_CHARS` and define it internally to this function. The error message now has hardcoded `63` which could also be interpolated with the variable.

Alternatively, we could pass the validation parameters as input

Pseudo-code:

```go
type namespaceValidation struct {
  max_allowed_chars int
  nameValidationRegex string
}

&namespaceValidation{
  max_allowed_chars: 63,
  nameValidationRegex: "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$",
}

func validateNamespace(validation namespaceValidation, namespace, object string) error {
```

This would also make testing more reliable